libenv/sysinfo.c

```diff
@@ -549,13 +549,9 @@ static void GetNameInfo3(EvalContext *ctx)
     bool found = false;
     for (i = 0; !found && (i < PLATFORM_CONTEXT_MAX); i++)
     {
```

Comment:
It would comfort me if we also test the assumption that `VSYSNAME.sysname` is in fact already lower case.
```suggestion
    {
#ifndef NDEBUG
        for (const char *ch = VSYSNAME.sysname; *ch != '\0'; ch++)
        {
            assert(islower(*ch));
        }
#endif /* NDEBUG */
```